lib/SIL/SILDeclRef.cpp

```diff
@@ -162,8 +162,22 @@ static unsigned getNaturalUncurryLevelAndKind(ValueDecl *vd,
     (void)var;
     return 0;
   }
-  default:
-    llvm_unreachable("unhandled kind of ValueDecl for SILDeclRef");
+
+  case DeclKind::GenericTypeParam:
+  case DeclKind::AssociatedType:
+  case DeclKind::TypeAlias:
+  case DeclKind::Enum:
+  case DeclKind::Struct:
+  case DeclKind::Protocol:
+  case DeclKind::Module:
+  case DeclKind::Param:
+  case DeclKind::Subscript:
+    llvm_unreachable("invalid kind of ValueDecl for SILDeclRef");
+
+#define VALUE_DECL(...)
+#define DECL(Id, Parent) case DeclKind::Id:
+#include "swift/AST/DeclNodes.def"
+    llvm_unreachable("impossible value decl");
```

Comment:
Question. How many of these are you including? By using the include file, you are not going to get the covered switch warning. I think you really need to handle /all/ cases directly. The whole point is if someone adds something to the DeclNodes.def file, we want them to use the covered switch warning to think about what needs to be done here.